code/modules/atmospherics/machinery/components/unary_devices/passive_vent.dm

```diff
@@ -24,26 +24,35 @@
 
 	var/datum/gas_mixture/external = loc.return_air()
 	var/datum/gas_mixture/internal = airs[1]
-	var/external_pressure = external.return_pressure()
-	var/internal_pressure = internal.return_pressure()
-	var/pressure_delta = abs(external_pressure - internal_pressure)
-
-	if(pressure_delta > 0.5)
-		if(external_pressure < internal_pressure)
-			var/air_temperature = (external.temperature > 0) ? external.temperature : internal.temperature
-			var/transfer_moles = (pressure_delta * external.volume) / (air_temperature * R_IDEAL_GAS_EQUATION)
-			var/datum/gas_mixture/removed = internal.remove(transfer_moles)
-			external.merge(removed)
-		else
-			var/air_temperature = (internal.temperature > 0) ? internal.temperature : external.temperature
-			var/transfer_moles = (pressure_delta * internal.volume) / (air_temperature * R_IDEAL_GAS_EQUATION)
-			transfer_moles = min(transfer_moles, external.total_moles() * internal.volume / external.volume)
-			var/datum/gas_mixture/removed = external.remove(transfer_moles)
-			if(isnull(removed))
-				return
-			internal.merge(removed)
-
-		active = TRUE
+
+	//is there a better way to do this in byond i.e. get the keys?
+	var/list/gas_list = new()
+	for(var/gas_id in internal.gases)
+		gas_list += gas_id
+	for(var/gas_id in external.gases)
+		gas_list += gas_id
+	gas_list = uniqueList(gas_list)
```

Comment:
If you know that uniqueList() make this why you don't do it in this way?
https://github.com/tgstation/tgstation/blob/528fc99f28efbb029060218bcd54a89e09fd4db5/code/__HELPERS/_lists.dm#L293-L297
```
	var/list/gas_list = new()
	for(var/gas_id in internal.gases)
		gas_list |= gas_id
	for(var/gas_id in external.gases)
		gas_list |= gas_id
```
and in general it is better to do it this way
```
	var/list/gas_list = new()
	gas_list |= internal.gases
	gas_list |= external.gases
```

Reply:
Didn't know realise and wasn't cognizant of how uniquelist does it.